### PR TITLE
Raspberry Pi Pico

### PR DESCRIPTION
**Pull request by alexandruradovici**
This pull request adds the Raspberry Pi Pico board description to `board_interface.py`.

### CODE REVIEW

File: tockloader/board_interface.py [hunk 1]
```diff
@@ -213,6 +213,14 @@ class BoardInterface:
                            flash bank $_CHIPNAME.flash nrf51 0x00000000 0 1 1 $_TARGETNAME;",
             },
         },
+        'raspberry_pi_pico': {'description': 'Raspberry Pi Pico',
+		                'arch': 'cortex-m0',
+		                'apps_start_address': 0x10020000,
```
Comment (bradjc) [inline tockloader/board_interface.py:218]: This now goes in tockloader.py.

File: tockloader/board_interface.py [hunk 2]
```diff
@@ -213,6 +213,14 @@ class BoardInterface:
...
+		                'page_size': 4096,
+		                'openocd': 'external',
+		                'openocd_prefix': 'source [find interface/raspberrypi-swd.cfg]; \
+		                                   source [find target/rp2040.cfg];'
```
Comment (bradjc) [inline tockloader/board_interface.py:222]: This is now

```
"openocd": {"prefix": "..."}
```

**Review by bradjc — COMMENTED** [review]
Sorry, cleaning up tockloader after it had kind of evolved piece meal means how boards are specified has changed.

File: tockloader/board_interface.py (hunk 1 above)
Comment (alexandruradovici) [reply tockloader/board_interface.py]: Sorry about this, I did not notice the change.

File: tockloader/tockloader.py [hunk 3]
```diff
@@ -106,6 +106,7 @@ class TockLoader:
             "microbit_v2": {"start_address": 0x00040000},
             "stm32f3discovery": {"start_address": 0x08020000},
             "stm32f4discovery": {"start_address": 0x08040000},
+            "raspberry_pi_pico": {'apps_start_address': 0x10020000},
```
Comment (bradjc) [inline tockloader/tockloader.py:109]:
```suggestion
            "raspberry_pi_pico": {"start_address": 0x10020000},
```

Comment (bradjc) [reply tockloader/tockloader.py:109]: ping

**Review by bradjc — APPROVED** [review]

**Pull request merged**